Build WebKit process extensions for all SDK variants

pvollan opened the pull request:
#### 6a907ad049edd443daa9236b711db5502ea1d508
<pre>
Build WebKit process extensions for all SDK variants
<a href="https://bugs.webkit.org/show_bug.cgi?id=271315">https://bugs.webkit.org/show_bug.cgi?id=271315</a>
<a href="https://rdar.apple.com/125062966">rdar://125062966</a>

Reviewed by Chris Dumez.

* Source/WebKit/Configurations/BaseExtension.xcconfig:

Canonical link: <a href="https://commits.webkit.org/276415@main">https://commits.webkit.org/276415@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/cc9e52275b8f96c6b0cf22e5288a24c63d3564bb

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/44592 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/23669 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/47042 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/47247 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/40590 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/27708 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/21065 "Built successfully") | [💥 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/36690 "An unexpected error occured. Recent messages:Printed configuration; Running apply-patch; Checked out pull request; Running run-layout-tests-in-stress-mode; Exiting early after 500 failures. 500 tests run. 1 flakes 500 failures; Uploaded test results (exception)") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/45168 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/20740 "Passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/38398 "Build is in progress. Recent messages:OS: Monterey (12.6.8), Xcode: 13.4.1; Skipping applying patch since patch_id isn't provided; Checked out pull request; Running run-api-tests") | [![loading-orange](https://github-production-user-asset-6210df.s3.amazonaws.com/3098702/291015173-08c448be-ac0a-4fd6-92a3-8165057445b7.png) 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/17743 "Build is in progress. Recent messages:Printed configuration; Running apply-patch; Checked out pull request; 3 api tests failed or timed out; Running re-run-api-tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/18207 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/2638 "Built successfully") | 
| [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/37769 "Build is in progress. Recent messages:") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/40824 "Build is in progress. Recent messages:OS: Sonoma (14.3.1), Xcode: 15.2; Skipping applying patch since patch_id isn't provided; Checked out pull request; Running run-api-tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/39807 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/48879 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/19549 "Built successfully") | [💥 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/16103 "An unexpected error occured. Recent messages:OS: Sonoma (14.3.1), Xcode: 15.2; Running apply-patch; Checked out pull request; Skipped layout-tests; Ignored 1 pre-existing failure based on results-db; Uploaded test results (exception)") | [💥 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/43632 "An unexpected error occured. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request; Skipped layout-tests; 92 flakes 47 failures 1 missing results; Uploaded test results (exception)") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/20880 "Built successfully") | | [![loading-orange](https://github-production-user-asset-6210df.s3.amazonaws.com/3098702/291015173-08c448be-ac0a-4fd6-92a3-8165057445b7.png) 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/42379 "Build is in progress. Recent messages:Printed configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request; run-api-tests (failure); 2 api tests failed or timed out; Running compile-webkit-without-change") | 
| | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/21211 "Build is in progress. Recent messages:OS: Sonoma (14.3.1), Xcode: 15.2; Running apply-patch; Checked out pull request; Running compile-webkit") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/6149 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/20546 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

bls1999 commented on Source/WebKit/Configurations/BaseExtension.xcconfig:
```diff
@@ -27,33 +27,29 @@ CLANG_ENABLE_OBJC_ARC = YES;
+SWIFT_OBJC_BRIDGING_HEADER[sdk=iphone*17.0*] = ;
+SWIFT_OBJC_BRIDGING_HEADER[sdk=iphone*17.1*] = ;
+SWIFT_OBJC_BRIDGING_HEADER[sdk=iphone*17.2*] = ;
+SWIFT_OBJC_BRIDGING_HEADER[sdk=iphone*17.3*] = ;
```
Possibly a dumb question, but does that need this line here?
`SWIFT_OBJC_BRIDGING_HEADER[sdk=iphone*17.4*] = ;`

bls1999 commented on Source/WebKit/Configurations/BaseExtension.xcconfig:
```diff
@@ -27,33 +27,29 @@ CLANG_ENABLE_OBJC_ARC = YES;
+EXCLUDED_SOURCE_FILE_NAMES[sdk=iphone*17.0*] = *;
+EXCLUDED_SOURCE_FILE_NAMES[sdk=iphone*17.1*] = *;
+EXCLUDED_SOURCE_FILE_NAMES[sdk=iphone*17.2*] = *;
+EXCLUDED_SOURCE_FILE_NAMES[sdk=iphone*17.3*] = *;
```
...and this line here?
`EXCLUDED_SOURCE_FILE_NAMES[sdk=iphone*17.4*] = *;`

pvollan replied:
That is a good point. For this SDK version we do want to include these source files, so there shouldn't be any excluded source files.

Thanks for reviewing!

bls1999 replied:
No problem, thanks for your reply!

pvollan committed:
Build WebKit process extensions for all SDK variants
https://bugs.webkit.org/show_bug.cgi?id=271315
rdar://125062966

Reviewed by Chris Dumez.

* Source/WebKit/Configurations/BaseExtension.xcconfig:

Canonical link: https://commits.webkit.org/276415@main
